content/en/blog/2023/basic-connector/index.md

```diff
@@ -0,0 +1,574 @@
+    )
+
+
+    // schema for connector
+    type connectorImp struct {
+        config Config
+        metricsConsumer consumer.Metrics
+        logger *zap.Logger
+    }
+
+
+    // function to create a new connector
+    func newConnector(logger *zap.Logger, config component.Config) (*connectorImp, error) {
+        logger.Info("Building exampleconnector connector")
+        cfg := config.(*Config)
+
+
+        return &connectorImp{
+        config: *cfg,
+        logger: logger,
+        }, nil
+    }
+
+
+    // Capabilities implements the consumer interface.
+    func (c *connectorImp) Capabilities() consumer.Capabilities {
+      return consumer.Capabilities{MutatesData: false}
+    }
+
+
+    func (c *connectorImp) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+        countMetrics := pmetric.NewMetrics() // creates metrics struct
+        spansWithAttribute := 0
+        for i := 0; i < td.ResourceSpans().Len(); i++ {
+            resourceSpan := td.ResourceSpans().At(i)
+
+            for j := 0; j < resourceSpan.ScopeSpans().Len(); j++ {
+                scopeSpan := resourceSpan.ScopeSpans().At(j)
+
+                for k := 0; k < scopeSpan.Spans().Len(); k++ {
+                    span := scopeSpan.Spans().At(k)
+                    attrs := span.Attributes()
+                    mapping := attrs.AsRaw()
+                    // counts the number of spans with the a specific attribute key
+                    for key, _ := range mapping {
+                        if key == c.config.AttributeName {
+                            spansWithAttribute++
+                        }
+                    }
+                }
+            }
+        }
+        // only create capacity for spans with the desired attribute
+        countMetrics.ResourceMetrics().EnsureCapacity(spansWithAttribute)
+        return c.metricsConsumer.ConsumeMetrics(ctx, countMetrics)
+    }
+
+
+    // Start implements the component.Component interface.
+    func (p *connectorImp) Start(ctx context.Context, _ component.Host) error {
+        p.logger.Info("Starting spanmetrics connector")
+        return nil
+    }
+
+    // Shutdown implements the component.Component interface.
+    func (p *connectorImp) Shutdown(context.Context) error {
+        p.logger.Info("Shutting down spanmetrics connector")
+        return nil
+    }
+
+&nbsp;<br>
+
+## Using the Component
+
+&nbsp;<br>
+
+### Summary of Using OpenTelemetry Collector Builder:
+
+You can use the OpenTelemetry Collector Builder tool to build your code and run
+it in a collector. The Collector Builder is a tool that enables you to build
+your own OpenTelemetry Collector binary. You can add or remove components
+(receivers, processors, connectors and exporters) to suit your needs.
+
+1.  Install OpenTelemetry Collector Builder: Install the CLI application using
+    the following command:
+
+        GO111MODULE=on go get github.com/open-telemetry/opentelemetry-collector-builder
```

Comment:
Instead of referring to a specific command here, let's just link to the builder documentation "Installation" section, so that we keep up to date:

```suggestion
1.  Follow the OpenTelemetry Collector Builder [installation instructions](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder#installation).
```